sklbench/benchs/sklearn_estimator.py

```diff
@@ -0,0 +1,405 @@
+        (estimator_class.__module__.startswith('daal4py') or
+         estimator_class.__module__.startswith('sklearnex'))
+    sklex_patching_stream = get_sklearnex_patching_stream()
+
+    metrics = dict()
+    estimator_instance = estimator_class(**estimator_params)
+    for stage in estimator_methods.keys():
+        for method in estimator_methods[stage]:
+            if hasattr(estimator_instance, method):
+                method_instance = getattr(estimator_instance, method)
+                if online_inference_mode and stage == 'inference':
+                    method_instance = create_online_function(method_instance, x_test)
+                metrics[method] = dict()
+                metrics[method]['time[ms]'], metrics[method]['time std[ms]'], _ = measure_case(
+                    bench_case, method_instance, *data_args[stage])
+                if ensure_sklearnex_patching:
+                    full_method_name = f'{estimator_class.__name__}.{method}'
+                    sklex_patching_stream.seek(0)
+                    method_is_patched = verify_patching(
+                        sklex_patching_stream, full_method_name)
+                    if not method_is_patched:
+                        logger.warning(f'{full_method_name} was not patched by sklearnex.')
+
+    quality_metrics = {
+        'training': get_subset_metrics_of_estimator(
+            task, 'training', estimator_instance, x_train, y_train),
+        'inference': get_subset_metrics_of_estimator(
+            task, 'inference', estimator_instance, x_test, y_test)
+    }
+    for method in metrics.keys():
+        for stage in estimator_methods.keys():
+            if method in estimator_methods[stage]:
+                metrics[method].update(quality_metrics[stage])
+
+    return metrics, estimator_instance
+
+
+def main(bench_case: BenchCase, filters: List[BenchCase]):
+    # get estimator class and ML task
+    library_name = get_bench_case_value(bench_case, 'algorithm:library')
+    estimator_name = get_bench_case_value(bench_case, 'algorithm:estimator')
+
+    estimator_class = get_estimator(library_name, estimator_name)
+    task = estimator_to_task(estimator_name)
+
+    # load and transform data
+    data, data_description = load_data(bench_case)
+    (x_train, x_test, y_train, y_test), data_description = split_and_transform_data(
+        bench_case, data, data_description)
+
+    # assign special values
+    assign_case_special_values_on_run(
+        bench_case, x_train, y_train, x_test, y_test, data_description)
+
+    # get estimator parameters
+    estimator_params = get_bench_case_value(bench_case, 'algorithm:estimator_params', dict())
+
+    # get estimator methods for measurement
+    estimator_methods = get_estimator_methods(bench_case)
+    online_inference_mode = get_bench_case_value(
+        bench_case, 'algorithm:online_inference_mode', False)
+
+    # benchmark case filtering
+    if not bench_case_filter(bench_case, filters):

```

Comment:
To be honest, having filtering out occurring while in operation was confusing. I think an initial filtering before tqdm in runner.py would  be better. This would give a more accurate representation of the cases to be run and the operational steps. I don't think it would cost much in terms of initial overhead. Unless log level is set to warning, it was hard to understand as I was trying to remove gpu cases (it seemed like they were running, but they should have been popped from the list).